spec/git-repository-async-spec.js

```diff
@@ -0,0 +1,598 @@
+'use babel'
+
+import fs from 'fs-plus'
+import path from 'path'
+import temp from 'temp'
+import Git from 'nodegit'
+
+import {it, beforeEach, afterEach} from './async-spec-helpers'
+
+import GitRepositoryAsync from '../src/git-repository-async'
+import Project from '../src/project'
+
+temp.track()
+
+function openFixture (fixture) {
+  return GitRepositoryAsync.open(path.join(__dirname, 'fixtures', 'git', fixture))
+}
+
+function copyRepository () {
+  let workingDirPath = temp.mkdirSync('atom-working-dir')
+  fs.copySync(path.join(__dirname, 'fixtures', 'git', 'working-dir'), workingDirPath)
+  fs.renameSync(path.join(workingDirPath, 'git.git'), path.join(workingDirPath, '.git'))
+  return fs.realpathSync(workingDirPath)
+}
+
+describe('GitRepositoryAsync', () => {
+  let repo
+
+  afterEach(() => {
+    if (repo != null) repo.destroy()
+  })
+
+  describe('@open(path)', () => {
+    it('repo is null when no repository is found', async () => {
+      repo = GitRepositoryAsync.open(path.join(temp.dir, 'nogit.txt'))
+
+      let threw = false
+      try {
+        await repo.repoPromise
+      } catch (e) {
+        threw = true
+      }
+
+      expect(threw).toBeTruthy()
+      expect(repo.repo).toBe(null)
+    })
+  })
+
+  describe('.getPath()', () => {
+    it('returns the repository path for a repository path', async () => {
+      repo = openFixture('master.git')
+      const repoPath = await repo.getPath()
+      expect(repoPath).toBe(path.join(__dirname, 'fixtures', 'git', 'master.git'))
+    })
+  })
+
+  describe('.isPathIgnored(path)', () => {
+    beforeEach(() => {
+      repo = openFixture('ignore.git')
+    })
+
+    it('resolves true for an ignored path', async () => {
+      const ignored = await repo.isPathIgnored('a.txt')
+      expect(ignored).toBeTruthy()
```

Comment:
Might be worth making these a little "stronger" with `toBe(true)` since this API should always return a `boolean` right? Looks like the `isStatusModified` specs do that currently.